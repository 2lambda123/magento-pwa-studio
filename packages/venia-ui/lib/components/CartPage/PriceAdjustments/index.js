export { default } from './priceAdjustments';

export { giftOptionsResolvers } from './GiftOptions';
