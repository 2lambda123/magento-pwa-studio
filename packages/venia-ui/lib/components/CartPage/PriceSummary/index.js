export { default, PriceSummaryQuery } from './priceSummary';
