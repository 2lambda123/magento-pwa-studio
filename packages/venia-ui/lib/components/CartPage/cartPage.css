.root {
    padding-top: 1.5rem;
    max-width: 64rem;
    margin: 0 auto;
}

.body {
    display: grid;
    gap: 2rem;
    grid-template-columns: 66% auto;
    /* The summary grid item spans the entire right column. */
    grid-template-areas:
        'items summary'
        'adjustments summary'
        'recents summary';
}

.heading_container {
    display: flex;
    align-items: baseline;
    justify-content: space-between;

    margin-bottom: 1rem;
}

.heading {
    line-height: 1.25em;
}

.items_container {
    grid-area: items;

    /* TEMPORARY STYLES FOR SCROLLABLE SUMMARY, CAN BE REMOVED */
    height: 900px;
    border: 1px dotted red;
}

.price_adjustments_container {
    grid-area: adjustments;
}

.recently_viewed_container {
    grid-area: recents;
}

.sign_in {
    font-weight: bold;
    text-decoration: underline;
}

.summary_container {
    grid-area: summary;
    height: 100%;
}

.summary_contents {
    position: sticky;
    /*
     * TODO: Use CSS Properties (variables) or something instead of hardcoding this.
     *  - 3.5 rem = min-height of nav header. See the "toolbar" class in header.css.
     *  - 1 rem = Spacing just for the sticky container.
     */
    top: calc(3.5rem + 1rem);
}

/*
 * Mobile-specific styles.
 */

@media (max-width: 640px) {
    .body {
        /* Only one column in mobile view. */
        grid-template-columns: 100%;
        grid-template-areas:
            'items'
            'adjustments'
            'summary';
    }

    .recently_viewed_container {
        display: none;
    }
}
