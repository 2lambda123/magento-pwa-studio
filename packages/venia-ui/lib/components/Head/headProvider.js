import React, { useEffect } from 'react';
import { HeadProvider } from 'react-head';

const removeExistingTitleTags = titleTags => {
    titleTags.forEach(titleTag => {
        if (titleTag) titleTag.parentNode.removeChild(titleTag);
    });
};

const VeniaHeadProvider = props => {
    useEffect(() => {
        const titleTagsToRemove = document.getElementsByTagName('title');
        removeExistingTitleTags(Array.from(titleTagsToRemove));
    }, []);

    return <HeadProvider>{props.children}</HeadProvider>;
};

export default VeniaHeadProvider;
