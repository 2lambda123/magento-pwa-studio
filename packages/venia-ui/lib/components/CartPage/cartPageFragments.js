import gql from 'graphql-tag';

import { ProductListingFragment } from './ProductListing/productListingFragments';
import { PriceSummaryFragment } from './PriceSummary/priceSummaryFragments';

export const CartPageFragment = gql`
    fragment CartPageFragment on Cart {
        id
        total_quantity
        ...ProductListingFragment
        ...PriceSummaryFragment
    }
    ${ProductListingFragment}
    ${PriceSummaryFragment}
`;
