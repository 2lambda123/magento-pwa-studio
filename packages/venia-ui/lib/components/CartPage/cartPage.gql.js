import { gql } from '@apollo/client';
import { CartPageFragment } from './cartPageFragments.gql';

export const GET_CART_DETAILS = gql`
    query getCartDetails($cartId: String!) {
        cart(cart_id: $cartId) {
            id
            ...CartPageFragment
        }
    }
    ${CartPageFragment}
`;
