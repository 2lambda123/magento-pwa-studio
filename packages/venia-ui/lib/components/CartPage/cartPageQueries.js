import gql from 'graphql-tag';
import { CartPageFragment } from './cartPageFragments';

export const GET_CART_DETAILS = gql`
    query getCartDetails($cartId: String!) {
        cart(cart_id: $cartId) {
            id
            ...CartPageFragment
        }
    }
    ${CartPageFragment}
`;
