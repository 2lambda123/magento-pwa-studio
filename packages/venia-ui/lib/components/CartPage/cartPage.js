import React, { useMemo } from 'react';

import { useCartPage } from '@magento/peregrine/lib/talons/CartPage/useCartPage';

import { Title } from '../../components/Head';
import Button from '../Button';

import PriceAdjustments from './PriceAdjustments';
import PriceSummary from './PriceSummary';
import ProductListing from './ProductListing';
import { mergeClasses } from '../../classify';
import defaultClasses from './cartPage.css';
import { GET_CART_DETAILS } from './cartPageQueries';

const CartPage = props => {
    const talonProps = useCartPage({
        queries: {
            getCartDetails: GET_CART_DETAILS
        }
    });

    const {
        handleSignIn,
        hasItems,
        isSignedIn,
        isCartUpdating,
        setIsCartUpdating
    } = talonProps;

    const classes = mergeClasses(defaultClasses, props.classes);

    const signInDisplay = useMemo(() => {
        return !isSignedIn ? (
            <Button
                className={classes.sign_in}
                onClick={handleSignIn}
                priority="high"
            >
                {'Sign In'}
            </Button>
        ) : null;
    }, [classes.sign_in, handleSignIn, isSignedIn]);

    const productListing = hasItems ? (
        <ProductListing setIsCartUpdating={setIsCartUpdating} />
    ) : (
        <h3>There are no items in your cart.</h3>
    );

    const priceAdjustments = hasItems ? (
        <PriceAdjustments setIsCartUpdating={setIsCartUpdating} />
    ) : null;
    const priceSummary = hasItems ? (
        <PriceSummary isCartUpdating={isCartUpdating} />
    ) : null;

    return (
        <div className={classes.root}>
            <Title>{`Cart - ${STORE_NAME}`}</Title>
            <div className={classes.heading_container}>
                <h1 className={classes.heading}>Cart</h1>
                {signInDisplay}
            </div>
            <div className={classes.body}>
                <div className={classes.items_container}>{productListing}</div>
                <div className={classes.price_adjustments_container}>
                    {priceAdjustments}
                </div>
                <div className={classes.summary_container}>
                    <div className={classes.summary_contents}>
                        {priceSummary}
                    </div>
                </div>
                <div className={classes.recently_viewed_container}>
                    <a href="https://jira.corp.magento.com/browse/PWA-270">
                        Recently Viewed to be completed by PWA-270.
                    </a>
                </div>
            </div>
        </div>
    );
};

export default CartPage;
