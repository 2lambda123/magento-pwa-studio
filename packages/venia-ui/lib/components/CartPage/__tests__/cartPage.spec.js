import React from 'react';
import ReactDOM from 'react-dom';
import { createTestInstance } from '@magento/peregrine';
import { useLazyQuery } from '@apollo/react-hooks';

import CartPage from '../cartPage';
import { HeadProvider } from '../../Head';

jest.mock('../PriceAdjustments', () => 'PriceAdjustments');
jest.mock('../PriceSummary', () => 'PriceSummary');
jest.mock('../ProductListing', () => 'ProductListing');

jest.mock('@apollo/react-hooks', () => {
    const runQuery = jest.fn();
    const queryResult = {
        data: null,
        error: null,
        loading: false
    };
    const useLazyQuery = jest.fn(() => [runQuery, queryResult]);

    return { useLazyQuery };
});

jest.mock('@magento/peregrine/lib/context/app', () => {
    const api = {
        toggleDrawer: jest.fn()
    };
    const state = {};
    const useAppContext = jest.fn(() => [state, api]);

    return { useAppContext };
});

jest.mock('@magento/peregrine/lib/context/cart', () => {
    const api = {};
    const state = { cartId: 'cart123' };
    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

jest.mock('@magento/peregrine/lib/context/user', () => {
    const api = {};
    const state = { isSignedIn: false };
    const useUserContext = jest.fn(() => [state, api]);

    return { useUserContext };
});

beforeAll(() => {
    /**
     * Mocking ReactDOM.createPortal because of incompatabilities
     * between ReactDOM and react-test-renderer.
     *
     * More info: https://github.com/facebook/react/issues/11565
     */
    ReactDOM.createPortal = jest.fn(element => {
        return element;
    });
});

afterAll(() => {
    ReactDOM.createPortal.mockClear();
});

test('renders empty cart text (no adjustments, list or summary) if cart is empty', () => {
    useLazyQuery.mockReturnValueOnce([
        jest.fn(),
        {
            data: {
                cart: {
                    total_quantity: 0
                }
            }
        }
    ]);

    const instance = createTestInstance(
        <HeadProvider>
            <CartPage />
        </HeadProvider>
    );
    expect(document.getElementsByTagName('title')[0].innerHTML).toBe(
        'Cart - Venia'
    );
    expect(instance.toJSON()).toMatchSnapshot();
});

test('renders components if cart has items', () => {
    useLazyQuery.mockReturnValueOnce([
        jest.fn(),
        {
            data: {
                cart: {
                    total_quantity: 1
                }
            }
        }
    ]);

    const instance = createTestInstance(
        <HeadProvider>
            <CartPage />
        </HeadProvider>
    );
    expect(document.getElementsByTagName('title')[0].innerHTML).toBe(
        'Cart - Venia'
    );
    expect(instance.toJSON()).toMatchSnapshot();
});
