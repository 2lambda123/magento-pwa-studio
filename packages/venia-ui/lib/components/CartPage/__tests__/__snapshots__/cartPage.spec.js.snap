// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components if cart has items 1`] = `
<div>
  <title>
    Cart - Venia
  </title>
  <div>
    <h1>
      Cart
    </h1>
    <button
      onClick={[Function]}
      type="button"
    >
      <span>
        Sign In
      </span>
    </button>
  </div>
  <div>
    <div>
      <ProductListing
        setIsCartUpdating={[Function]}
      />
    </div>
    <div>
      <PriceAdjustments
        setIsCartUpdating={[Function]}
      />
    </div>
    <div>
      <div>
        <PriceSummary
          isUpdating={false}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders empty cart text (no adjustments, list or summary) if cart is empty 1`] = `
<div>
  <title>
    Cart - Venia
  </title>
  <div>
    <h1>
      Cart
    </h1>
    <button
      onClick={[Function]}
      type="button"
    >
      <span>
        Sign In
      </span>
    </button>
  </div>
  <div>
    <div>
      <h3>
        There are no items in your cart.
      </h3>
    </div>
    <div />
    <div>
      <div />
    </div>
  </div>
</div>
`;
